### PR TITLE
OpenZeppelin updated ERC-721

### PR DESCRIPTION
**Pull request by jamesmorgan**
* [x] bring over new erc-721 standard
* [x] bring over tests
* [x] Does the app still work?
* [ ] Add missing tests for new functionality

### CODE REVIEW

**Comment by andygray** [comment]
looking good - need to get my head round the receiver and "safe" transfers

File: src/router/index.js [hunk 1]
```diff
@@ -46,20 +51,38 @@ export default new Router({
...
     },
     {
-      path: '/:artistCode/:edition/:tokenId',
+      path: '/editions/:edition',
+      name: 'confirmPurchase',
+      component: ConfirmPurchase,
+      props: true
+    },
+    {
+      path: '/artists/:artistCode/editions/:edition/assets/:tokenId',
```
Comment (jamesmorgan) [inline src/router/index.js]: Should this not `/edition/` without the `(s)` along with `/asset/`? - I like it though 👍

File: src/router/index.js [hunk 2]
```diff
@@ -46,20 +51,38 @@ export default new Router({
       name: 'license',
       component: License
     },
+    // TODO? complete the REST Api?
+    // {
+    //   path: '/assets',
+    //   name: 'assets',
+    //   component: Assets
+    // },
```
Comment (jamesmorgan) [inline src/router/index.js]: This could also route to gallery?

File: src/router/index.js [hunk 3]
```diff
@@ -53,13 +53,36 @@ export default new Router({
...
+    },
+    {
+      path: '/editions/:edition',
       name: 'confirmPurchase',
       component: ConfirmPurchase,
       props: true
     },
     {
-      path: '/edition/:edition/:tokenId',
+      path: '/artists/:artistCode/editions/:edition/assets/:tokenId',
```
Comment (jamesmorgan) [inline src/router/index.js]: Should this not `/edition/` without the `(s)` along with `/asset/`? - I like it though 👍

File: src/router/index.js [hunk 4]
```diff
@@ -53,13 +53,36 @@ export default new Router({
...
+      name: 'completePurchase',
+      component: CompletePurchase,
+      props: true
+    },
+    {
+      path: '/assets/:tokenId',
       name: 'completePurchase',
       component: CompletePurchase,
```
Comment (jamesmorgan) [inline src/router/index.js]: We should probably sort this file so it its in a better order

File: src/store/artist-data.js [hunk 5]
```diff
@@ -1,72 +1,72 @@
...
     img: 'static/artists/BarrieJD_profile.png',
     featured: true
   },
   {
-    id: 'JOC',
+    artistCode: 'JOC',
     name: 'James O’Connell',
     twitter: '@Jamesp0p',
-    bio: 'James Oconnell is a Manchester based creative who has a passion for mixing colour and lines. He applies his minimalistic style to a variety of themes and has created work for the likes of Wired Magazine, Youtube, T3 Magazine, BBC and The Atlantic to name a few. He is hugely inspired by contemporary pop culture in all its forms - from motion pictures, music to advertising and sport. Simplicity is the key to his work allowing the expression of the idea to reign supreme.',
+    bio: 'James Oconnell is a Manchester based creative who has a passion for mixing colour and lines. He applies his minimalistic style to a variety of themes and has created work for the likes of Wired Magazine, Youtube, T3 Magazine, BBC and The Atlantic to name a few. He is hugely inspired by contemporary pop culture in all its forms - from motion pictures, music to advertising and sport. Simplicity is the key to his work allowing the expression of the artistCodeea to reign supreme.',
```
Comment (jamesmorgan) [inline src/store/artist-data.js]: looks like a find'n'replace over the `id` has changed lots of text

File: contracts/Migrations.sol [hunk 6]
```diff
@@ -1,23 +0,0 @@
-pragma solidity ^0.4.4;
```
Comment (jamesmorgan) [inline contracts/Migrations.sol]: Reinstate file as deleted by accident

File: test/helpers/latestTime.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+// Returns the time of the last mined block in seconds
```
Comment (jamesmorgan) [inline test/helpers/latestTime.js]: remove unneeded helpers once †ests passing

**Pull request merged**